lib/pbench/server/api/resources/query_apis/iteration_samples.py

```diff
@@ -0,0 +1,240 @@
+        measurement type and measurement title
+
+        Example:
+        {
+        "controller": {
+            "doc_count_error_upper_bound": 0,
+            "sum_other_doc_count": 0,
+            "buckets": [
+                {
+                    "key": "Controller_name",
+                    "doc_count": 50
+                }
+            ]
+        },
+        "name": {
+            "doc_count_error_upper_bound": 0,
+            "sum_other_doc_count": 0,
+            "buckets": [
+                {
+                    "key": "Dataset_name",
+                    "doc_count": 50
+                }
+            ]
+        },
+        "id": {
+            "doc_count_error_upper_bound": 0,
+            "sum_other_doc_count": 0,
+            "buckets": [
+                {
+                    "key": "f3a37c9891a78886639e3bc00e3c5c4e",
+                    "doc_count": 50,
+                    "type": {
+                        "doc_count_error_upper_bound": 0,
+                        "sum_other_doc_count": 0,
+                        "buckets": [
+                            {
+                                "key": "throughput",
+                                "doc_count": 50,
+                                "title": {
+                                    "doc_count_error_upper_bound": 0,
+                                    "sum_other_doc_count": 0,
+                                    "buckets": [
+                                        {
+                                            "key": "Gb_sec",
+                                            "doc_count": 50,
+                                            "uid": {
+                                                "doc_count_error_upper_bound": 0,
+                                                "sum_other_doc_count": 0,
+                                                "buckets": [
+                                                    {
+                                                        "key": "client_hostname:127.0.0.1-server_hostname:127.0.0.1-server_port:20010",
+                                                        "doc_count": 25
+                                                    },
+                                                    {
+                                                        "key": "client_hostname:all-server_hostname:all-server_port:all",
+                                                        "doc_count": 25
+                                                    }
+                                                ]
+                                            }
+                                            }
+                                        ]
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                ]
+            }
+        }
+        """
+        return jsonify(es_json["aggregations"])
```

Comment:
And, again, we need to return the `hits` as well as the aggregations. If we're not including the model effect postprocessing, we probably should just return the raw Elasticsearch exactly as the dashboard service layer does now.

Reply:
We _should_ be trying to insulate the client from the details of our server implementation (e.g., our choice to use Elasticsearch)...subject to all of our other constraints (such as compatibility and ease of evolution), of course....